pyccel/stdlib/cwrapper/cwrapper.h

```diff
@@ -0,0 +1,207 @@
+{
+	return (int32_t)PyLong_AsLong(object);
+}
+//-----------------------------------------------------//
+inline int16_t	PyInt16_to_Int16(PyObject *object)
+{
+	return (int16_t)PyLong_AsLong(object);
+}
+//-----------------------------------------------------//
+inline int8_t	PyInt8_to_Int8(PyObject *object)
+{
+	return (int8_t)PyLong_AsLong(object);
+}
+//-----------------------------------------------------//
+inline bool	PyBool_to_Bool(PyObject *object)
+{
+	return object == Py_True;
+}
+//-----------------------------------------------------//
+inline float	PyFloat_to_Float(PyObject *object)
+{
+	return (float)PyFloat_AsDouble(object);
+}
+//-----------------------------------------------------//
+inline double	PyDouble_to_Double(PyObject *object)
+{
+	return PyFloat_AsDouble(object);
+}
+
+
+/*
+ * Functions : Cast functions
+ * ---------------------------
+ * Some of the function used below are based on C/python api
+ * with more tolerance to different precisions and complex type.
+ * Collect the python object from the C object
+ * Parameters :
+ *     object : the C object
+ *
+ * Returns    :
+ *     boolean : python object
+ */
+PyObject	*Complex128_to_PyComplex(double complex *c);
+PyObject	*Complex64_to_PyComplex(float complex *c);
+
+PyObject	*Bool_to_PyBool(bool *b);
+
+PyObject	*Int64_to_PyLong(int64_t *i);
+PyObject	*Int32_to_PyLong(int32_t *i);
+PyObject	*Int16_to_PyLong(int16_t *i);
+PyObject	*Int8_to_PyLong(int8_t *i);
+
+PyObject	*Double_to_PyDouble(double *d);
+PyObject	*Float_to_PyDouble(float *d);
+
+/*
+ * Functions : Type check functions
+ * ---------------------------
+ * Some of the function used below are based on C/python api and numpy/c api with
+ * more tolerance to different precisions, different system architectures and complex type.
+ * Check the C data type ob a python object
+ * Parameters :
+ *     object     : the python object
+ *
+ * Returns    :
+ *     boolean : logic statement responsible for checking python data type
+ *
+ * Reference of the used c/python api function
+ * ---------------------------------------------------
+ * https://docs.python.org/3/c-api/long.html#c.PyLong_Check
+ * https://docs.python.org/3/c-api/complex.html#c.PyComplex_Check
+ * https://docs.python.org/3/c-api/float.html#c.PyFloat_Check
+ * https://docs.python.org/3/c-api/bool.html#c.PyBool_Check
+ * https://numpy.org/doc/1.17/reference/c-api.array.html#c.PyArray_IsScalar
+ */
+//--------------------------------------------------------//
+inline bool    PyIs_Int8(PyObject *o)
+{
+    return PyArray_IsScalar(o, Int8);
+}
+//--------------------------------------------------------//
+inline bool    PyIs_Int16(PyObject *o)
+{
+    return PyArray_IsScalar(o, Int16);
+}
+//--------------------------------------------------------//
+inline bool    PyIs_Int32(PyObject *o)
+{
+#ifdef _WIN32
+    return PyLong_Check(o) || PyArray_IsScalar(o, Int32);
+#else
+    return PyArray_IsScalar(o, Int32);
+#endif
```

Comment:
I think you need to reverse this too.

Reply:
No. The original code was correct. It's just that before the code used `#ifdef` and `#ifndef` to keep a similar order for the commands. I found this confusing as I assumed that the top section was always windows so I changed it. The aim is to only use `PyLongCheck` if the precision matches the default precision (32 on windows, 64 on linux)

Reply:
Ah, okay.